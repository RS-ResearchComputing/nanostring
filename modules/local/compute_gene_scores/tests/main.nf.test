nextflow_process {

    name "Test Process COMPUTE_GENE_SCORES"
    script "../main.nf"
    process "COMPUTE_GENE_SCORES"

    tag "modules"
    tag "modules_nfcore"
    tag "compute_gene_scores"

    test("test profile") {

        setup {
            run("NACHO_NORMALIZE") {
                script "../../nacho/normalize.nf"
                process {
                    """
                    input[0] = Channel.of(
                        [
                            file(params.pipelines_testdata_base_path + 'nanostring/RCC_files/salmon_01_01.RCC', checkIfExists: true),
                            file(params.pipelines_testdata_base_path + 'nanostring/RCC_files/salmon_02_02.RCC', checkIfExists: true),
                            file(params.pipelines_testdata_base_path + 'nanostring/RCC_files/salmon_03_03.RCC', checkIfExists: true),
                            file(params.pipelines_testdata_base_path + 'nanostring/RCC_files/salmon_04_04.RCC', checkIfExists: true)
                        ]
                    )
                    input[1] = file(params.pipelines_testdata_base_path + 'nanostring/samplesheets/samplesheet_test.csv', checkIfExists: true)
                    """
                }
            }
        }
        when {
            process {
                """
                input[0] = NACHO_NORMALIZE.out.normalized_counts
                input[1] = file(params.pipelines_testdata_base_path + 'nanostring/gene_score_test.yaml', checkIfExists: true )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
