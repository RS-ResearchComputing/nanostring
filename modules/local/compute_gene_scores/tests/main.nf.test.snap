{
    "test profile": {
        "content": [
            {
                "0": [
                    [
                        "signature_scores_mqc.txt:md5,315dc730c79b02067b542074eaeb07b9",
                        "signature_scores_qc_mqc.txt:md5,88ec48e1d7488357663557b378615923"
                    ]
                ],
                "1": [
                    "versions.yml:md5,61fc461ef40beb8b95aa31c8ee4d6760"
                ],
                "scores_for_mqc": [
                    [
                        "signature_scores_mqc.txt:md5,315dc730c79b02067b542074eaeb07b9",
                        "signature_scores_qc_mqc.txt:md5,88ec48e1d7488357663557b378615923"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,61fc461ef40beb8b95aa31c8ee4d6760"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-19T18:06:40.93622974"
    }
}